spec/app/domain/authentication/authn_k8s/validate_pod_request_spec.rb

```diff
@@ -70,43 +70,88 @@
   end
 
   context "A ValidatePodRequest invocation" do
-    subject do
-      Authentication::AuthnK8s::ValidatePodRequest.new(
-        resource_class:                resource_class,
-        k8s_object_lookup_class:       k8s_object_lookup_class,
-        validate_security:             mocked_security_validator,
-        enabled_authenticators:        "#{authenticator_name}/#{service_id}",
-        validate_application_identity: validate_application_identity
-      ).call(
-        pod_request: pod_request
-      )
+    context "that passes webservice validations" do
+      subject do
+        Authentication::AuthnK8s::ValidatePodRequest.new(
+          resource_class:                      resource_class,
+          k8s_object_lookup_class:             k8s_object_lookup_class,
+          validate_user_can_access_webservice: mock_validate_user_can_access_webservice(validation_succeeded: true),
+          validate_webservice_is_whitelisted:  mock_validate_webservice_is_whitelisted(validation_succeeded: true),
+          enabled_authenticators:              "#{authenticator_name}/#{service_id}",
+          validate_application_identity:       validate_application_identity
+        ).call(
+          pod_request: pod_request
+        )
       end
 
-    it_behaves_like "raises an error when security validation fails"
+      it 'raises PodNotFound when pod is not known' do
+        allow(k8s_object_lookup_class).to receive(:pod_by_name)
+                                            .with(spiffe_name, spiffe_namespace)
+                                            .and_return(nil)
+
+        expect { subject }.to(
+          raise_error(
+            ::Errors::Authentication::AuthnK8s::PodNotFound
+          )
+        )
+      end
 
-    it 'raises PodNotFound when pod is not known' do
-      allow(k8s_object_lookup_class).to receive(:pod_by_name)
-                                          .with(spiffe_name, spiffe_namespace)
-                                          .and_return(nil)
+      it 'raises an error when application identity validation fails' do
+        allow(validate_application_identity).to receive(:call)
+                                                  .and_raise('FAKE_APPLICATION_IDENTITY_ERROR')
+
+        expect { subject }.to(
+          raise_error(
+            /FAKE_APPLICATION_IDENTITY_ERROR/
```

Comment:
In a previous test, ` /FAKE_APPLICATION_IDENTITY_ERROR/` was removed and replaced with `validate_webservice_is_whitelisted_error` which is 
`let(:validate_webservice_is_whitelisted_error) { "validate whitelisted webservice error" }`

so IMO it will look cleaner if we  choose a style.

Reply:
Done